test/end-to-end.js

```diff
@@ -1043,6 +1043,66 @@ describe('end-to-end', function() {
         .end(done);
     }
   });
+
+  describe('Multi-project workspace support', function() {
```

Comment:
I prefer consistency and have all workspace-related tests in the same file, that is `test/workspace.js`.

I agree we should eventually make more granular test files (`test/unit/workspace.js`, `test/integration/workspace.js`, etc.), but that's out of scope of this pull request.

Since `test/end-to-end` is not the right place for this new test, I strongly prefer to put it where other similar tests are already (`test/workspace.js`).